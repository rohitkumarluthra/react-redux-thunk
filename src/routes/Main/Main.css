.main {
    flex: 80%;
    /* background-color: #f1f1f1; */
    padding: 20px;
    height: 100vh;
}

.wrapper {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
}

.img {
  margin: 10px;
  background-color: #f1f1f1;
}

.btn_wrapper {
  margin: auto;
  width: 10%;
}

.btn_more {
  background-color: #4CAF50;
  border: none;
  color: white;
  padding: 10px 20px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  width: 100px;
  margin: 20px;
}