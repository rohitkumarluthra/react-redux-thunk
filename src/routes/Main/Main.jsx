import React from 'react';
import { connect } from 'react-redux';

import "./Main.css";
import Spinner from '../../components/Spinner/Spinner';
import { fetchSingleCategory } from '../../redux/categories/actions';

export class Main extends React.PureComponent {
    render() {
      const { images, fetching } = this.props.categories;

      return (
        <div className="main">
          <div className="wrapper">
            {this.renderImages(images)}
            <Spinner
              left={'60%'}
              top={'60%'}
              fontSize={'3em'}
              visible={fetching}
            />
          </div>
          <div className="btn_wrapper">
            <button className="btn_more" onClick={this.handleMoreClick}>
              More
            </button>
          </div>
        </div>
      );
    }

    componentDidMount() {
      const { limit, page, activeCategory } = this.props.categories;
      if (activeCategory !== -1) {
        this.props.getSingleCategory(activeCategory, limit, page, false);
      }
    }

    componentDidUpdate(prevProps) {
      const { limit, page, activeCategory } = this.props.categories;
      if (prevProps.categories.activeCategory !== activeCategory) {
        this.props.getSingleCategory(activeCategory, limit, page, false);
      }
    }

    ///////////////////////////////////////////////////////////////////////
    //  RENDER METHODS
    ///////////////////////////////////////////////////////////////////////
    renderImages(images) {
      return images.map((item, i) => {
        return (<img key={item.id + i} src={item.url} width={450} height={400} alt={item.id} className="img"></img>);
      });
    }

    ///////////////////////////////////////////////////////////////////////
    //  EVENT HANDLERS
    ///////////////////////////////////////////////////////////////////////
    handleMoreClick = () => {
      const { limit, page, activeCategory } = this.props.categories;
      this.props.getSingleCategory(activeCategory, limit, page + 1, true);
    }
}

///////////////////////////////////////////////////////////////////////
//  REDUX CONNECTION
///////////////////////////////////////////////////////////////////////
function mapDispatchToProps(dispatch) {
  return {
    getSingleCategory: (categoryId, limit, page, isMore) => dispatch(fetchSingleCategory(categoryId, limit, page, isMore)),
  };
}

function mapStateToProps(state) {
  const { categories } = state;
  const categoriesToJS = categories.toJS();

  return {
    categories : categoriesToJS.categories,
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Main);