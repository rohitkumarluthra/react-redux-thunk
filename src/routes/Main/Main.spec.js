import React from 'react';
import Main from './Main';
import '../../setupTests';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import { mount, shallow } from 'enzyme';

describe('Main renders',() => {
  it('renders', () => {
    const wrapper = shallow(<Main store={store} />);
    expect(wrapper.exists()).toBe(true);
  })
  it('Main component created', () => {
    const rendered = renderer.create(
      <Main store={store} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  })

  it('find more button', () => {
    const wrapper = mount(<Main store={store}/>);
    const text = wrapper.find('button').text();
    expect(text).toEqual('More');
  });
})