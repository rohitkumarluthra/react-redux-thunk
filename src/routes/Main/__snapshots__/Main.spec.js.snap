// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders Main component created 1`] = `
<div
  className="main"
>
  <div
    className="wrapper"
  >
    <div
      className="spinner"
      style={
        Object {
          "display": "none",
          "fontSize": "3em",
          "left": "60%",
          "top": "60%",
        }
      }
    >
      <span
        className="fa fa-circle-o-notch fa-spin"
      />
    </div>
  </div>
  <div
    className="btn_wrapper"
  >
    <button
      className="btn_more"
      onClick={[Function]}
    >
      More
    </button>
  </div>
</div>
`;
